src/extern.jl

```diff
@@ -12,12 +12,13 @@ Estimate probabilistic PCA model for noise that is heteroscedastic across sample
 + `iters`  : number of iterations to run
 + `init`   : initial model (will be modified in-place)
 + `vknown` : variances are known (do not update) default `false`
++ `varfloor` : minimum variance to prevent failing cases, default = 0
 Output is a [`HePPCATModel`](@ref) object.
 """
-function heppcat(Y,k,iters::Integer;init=homppca(Y,k),vknown::Bool=false)
+function heppcat(Y,k,iters::Integer;init=homppca(Y,k),vknown::Bool=false,varfloor::Real=0.0)
```

Comment:
Use `zero` for more type-stability.
```suggestion
function heppcat(Y,k,iters::Integer;init=homppca(Y,k),vknown::Bool=false,varfloor=zero(eltype(Y)))
```

Reply:
So I left this one as is since I got an error message. Y in this case is a list so I think we might want something like this instead zero(eltype(Y[1])) ???

Here is the error message for convenience when you do zero(eltype(Y)) :

MethodError: no method matching zero(::Type{Any})
Closest candidates are:
  zero(::Type{Union{Missing, T}}) where T at /opt/julia-1.7.2/share/julia/base/missing.jl:105
  zero(::Union{Type{P}, P}) where P<:Dates.Period at /opt/julia-1.7.2/share/julia/stdlib/v1.7/Dates/src/periods.jl:53
  zero(::T) where T<:Dates.TimeType at /opt/julia-1.7.2/share/julia/stdlib/v1.7/Dates/src/types.jl:450
  ...

Stacktrace:
 [1] zero(#unused#::Type{Any})
   @ Base ./missing.jl:106
 [2] top-level scope
   @ In[129]:1
 [3] eval
   @ ./boot.jl:373 [inlined]
 [4] include_string(mapexpr::typeof(REPL.softscope), mod::Module, code::String, filename::String)
   @ Base ./loading.jl:1196

Reply:
Oh right, try this instead.
```suggestion
function heppcat(Y,k,iters::Integer;init=homppca(Y,k),vknown::Bool=false,varfloor=zero(eltype(first(Y))))
```

Reply:
Should we plan ahead for units and use something like:
`zero(eltype(first(Y)[1]*first(Y)[1]))`
(I am unsure if `first(Y)` is a scalar so that is why i added `[1]`.)